### ISSUE TITLE
Hostile mobs fail to spawn in the end dimension on 1.18

### ISSUE THREAD

**Issue by Wesley1808** [Spottedleaf/Starlight#99]
Since commit https://github.com/PaperMC/Starlight/commit/440547d96828cc89e2ffaf8ea4b9f9406d4f80f7, enderman have not been able to spawn in the end dimension.
I have tested this on newer commits aswell, but that doesn't change anything.

The end world was also tested when completely regenerated, without any other mods installed - and reverting to an older commit afterwards seems to fix the issue aswell.

**Comment by Helinos** [comment]
I've been able to reproduce this issue.

**Comment by Helinos** [comment]
This issue seems to because caused by the new `get RawBrightness` method that was added/moved to [this class](https://github.com/PaperMC/Starlight/blob/fabric-dev/1.18/src/main/java/ca/spottedleaf/starlight/common/light/StarLightInterface.java)  in the commit. This method changed where it got the sky light value from, that being `getSkyLightValue()` as apposed to where it used to get it: `skyReader.getLightValue()`.
This change causes this issue because getting the skylight value from the former method has no check for whether the dimension actually _has_ a skylight, where the latter method does. This causes The End to have a sky light of 15 and subsequently prevent endermen from spawning, as they can only spawn in a light level below 12 or whatever.

I reverted this change in [my fork](https://github.com/DontStarve72/Starlight) and it seems to fix it.
